Formula/b/batt.rb

```diff
@@ -0,0 +1,43 @@
+class Batt < Formula
+  desc "Control and limit battery charging on Apple Silicon MacBooks"
+  homepage "https://github.com/charlie0129/batt"
+  url "https://github.com/charlie0129/batt/archive/refs/tags/v0.2.1.tar.gz"
+  sha256 "8602740c33da8aa96c473eba5751229c34264cade53bda905428abc472c55afd"
+  license "GPL-2.0-only"
+
+  depends_on "go" => :build
+  depends_on arch: :arm64        # Apple Silicon only
+  depends_on macos: :big_sur     # Works on Big Sur per 0.2.1 release note
+  uses_from_macos "curl"
+
+  def install
+    # Tool does not provide separate control for AllowNonRootAccess.
+    # Normally it's changed during service file installation (by it self),
+    # but I made a service block. Changing defaults here.
+    inreplace "conf.go", "AllowNonRootAccess:      false,",
+      "AllowNonRootAccess:      true,"
+    # Make populates version tags by querying git. Tarball won't provide these info.
+    system "make", "VERSION=v0.2.1", "GIT_COMMIT=90df4aaefb930ab05a6763c96866e8db0faf698e"
+    bin.install "bin/batt"
+    inreplace "hack/cc.chlc.batt.plist", "/path/to/batt", opt_bin/"batt"
+    system "plutil", "-insert", "ProcessType", "-string", "Background", "--",
+           "hack/cc.chlc.batt.plist"
+    prefix.install "hack/cc.chlc.batt.plist"
+  end
+
+  def caveats
+    <<~EOS
+      Note the service must be running before most of batt's commands will work.
+    EOS
+  end
+
+  service do
+    name macos: "cc.chlc.batt"
+    require_root true
+  end
+
+  test do
+    assert_match "v0.2.1 90df4aaefb930ab05a6763c96866e8db0faf698e",
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
The majority (short of help and version) of functionalities of this tool requires first running its daemon as root, which creates a socket under `/var/run` for communication. I guess it falls into the "cannot function without credentials" category then. I'll see if that makes a more meaningful test.